sdk/servicebus/Azure.Messaging.ServiceBus/stress/src/Program.cs

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using CommandLine;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Diagnostics.Tracing;
+using Azure.Core.Diagnostics;
+
+namespace Azure.Messaging.ServiceBus.Stress;
+
+/// <summary>
+///   The main program thread that allows for both local and deployed test runs.
+///   Determines which tests should be run and starts them in the background.
+/// </summary>
+///
+public class Program
+{
+    /// <summary>
+    ///   Parses the command line <see cref="Options" /> and runs the specified tests.
+    /// </summary>
+    ///
+    /// <param name="args">The command line inputs.</param>
+    ///
+    public static async Task Main(string[] args)
+    {
+        // Parse command line arguments
+        await CommandLine.Parser.Default.ParseArguments<Options>(args).WithParsedAsync(RunOptions).ConfigureAwait(false);
+    }
+
+    /// <summary>
+    ///   Starts a background task for each test and role that needs to be run in this process, and waits for all
+    ///   test runs to completed before finishing the run.
+    /// </summary>
+    ///
+    /// <param name="opts">The parsed command line inputs.</param>
+    ///
+    private static async Task RunOptions(Options opts)
+    {
+        // See if there are environment variables available to use in the .env file
+        var environment = new Dictionary<string, string>();
+        var environmentFile = Environment.GetEnvironmentVariable("ENV_FILE");
+        environment = EnvironmentReader.LoadFromFile(environmentFile);
+
+        environment.TryGetValue(EnvironmentVariables.ApplicationInsightsKey, out var appInsightsKey);
+        environment.TryGetValue(EnvironmentVariables.ServiceBusConnectionString, out var serviceBusConnectionString);
+
+        // If a job index is provided, a single role is started, otherwise, all specified roles within the
+        // test scenario runs are run in parallel.
+
+        var testParameters = new TestParameters();
```

Comment:
Is TestParameters disposable?

Reply:
Yeah, it holds the SHA instance to use to hash the bodies to check for any message corruption, dispose is called in https://github.com/Azure/azure-sdk-for-net/blob/6261184135d3dd9d38659e70dd4ce57d31193f89/sdk/servicebus/Azure.Messaging.ServiceBus/stress/src/Program.cs#L132